pages/index.tsx

```diff
@@ -1,156 +1,206 @@
+      errorFound = true;
+    }
+    if (errorFound) {
+      return;
+    }
+
+    const generatedUUID = crypto.randomUUID();
+    const JSON_Object: FullLocalStorage[] = [
+      {
+        name: newReceiver,
+        gift: newGiftName,
+        id: generatedUUID,
+        createdDate: new Date().getTime(),
+      },
+    ];
+    let localStorageGiftData: FullLocalStorage[] = JSON.parse(
+      getLocalStorage('giftData'),
+    );
+    localStorageGiftData = localStorageGiftData.concat(JSON_Object);
 
-    if (typeof giftName !== 'string' || giftName.length === 0)
-      throw new Error("Invalid gift's name!");
-    if (typeof receiverName !== 'string' || receiverName.length === 0)
-      throw new Error("Invalid receiver's name!");
+    setLocalStorage('giftData', JSON.stringify(localStorageGiftData));
+    setGiftData(localStorageGiftData);
+    setNewGiftName('');
+    setNewReceiver('');
+  }
 
-    const JSON_Object: { name: string; gift: string; keyID: string } = {
-      name: receiverName,
-      gift: giftName,
-      keyID: generateKeyID(),
-    };
+  function handleDeletion(gift: FullLocalStorage) {
+    const confirmDeletion = confirm(`Deleting ${gift.name} - ${gift.gift}`);
+    if (confirmDeletion) {
+      let localStorageGifts: FullLocalStorage[] = JSON.parse(
+        getLocalStorage('giftData'),
+      );
+      localStorageGifts = localStorageGifts.filter(
+        (localStorageGift) => localStorageGift.id !== gift.id,
+      );
+      setLocalStorage('giftData', JSON.stringify(localStorageGifts));
+      refreshGiftList();
+    }
+  }
 
-    setLocalStorage(generateID(), JSON.stringify(JSON_Object));
-    setGiftData((previousValue) => previousValue.concat(JSON_Object));
+  function refreshGiftList() {
+    const sortedGifts = sortGiftsOldestFirst(
+      JSON.parse(getLocalStorage('giftData')),
+    );
+    setGiftData(sortedGifts);
   }
 
   return (
-    <main className={`bg-white w-full max-w-full h-screen ${inter.className}`}>
-      <div id="fullScreenContainer" className="justify-center grid h-5">
-        <div id="formContainer" className="mt-5">
-          <form id="giftForm" className="" action={'/api/create'}>
-            <div id="formTitle" className="text-2xl pt-4">
-              Lahjalistaidea
-            </div>
-
-            <div id="giftNameContainer" className="pt-4 grid">
-              <label htmlFor="giftName">Lahja</label>
-              <input
+    <Main className={`bg-white w-full max-w-full h-screen ${inter.className}`}>
+      <Container className="justify-center grid h-5">
+        <Container className="mt-5">
+          <Form onSubmit={(e) => e.preventDefault()}>
+            <TitleText className="text-2xl pt-4">Lahjalistaidea</TitleText>
+            <Container className="pt-4 grid">
+              <Label htmlFor="giftName">Lahja</Label>
+              <Input
                 id="giftName"
+                required={true}
+                onChange={(event) => handleGiftNameChange(event)}
```

Comment:
Yksinkertaisempi laittaa inline tähän
```suggestion
                onChange={(event) => setNewGiftName(event.target.value)}
```